src/v/storage/log_manager.cc

```diff
@@ -756,10 +756,15 @@ ss::future<usage_report> log_manager::disk_usage() {
         logs.push_back(it.second->handle);
     }
 
+    ss::semaphore limit(20);
```

Comment:
nit: curious how this was picked, and whether we've considered making this configurable?

I guess the main effects of getting this wrong are:
- too low and disk usage becomes slower to refresh, and slower to respond to bursty workloads that may cause redpanda to run out of space
- too high and we flood the task queues

I guess both of these become more dramatic when there are a large number of partitions and segments, but it might be nice to be able to pick whichever's more tolerable case-by-case

Reply:
good point. my hunch is that that this value has very little affect on the ability for space management to keep up. the dominate cost that could be serialized is fstat for segments that don't a cached file size--recently compacted segments and active segments.

so i think that at a concurrency limit of 1 might have noticeable delay, the equivalence class of say like 20-200+ doesn't have much variance. then at 50,000+ futures queued up without any control starves other work.

anyway, i think i can pretty easily make this configurable. good thing to add tomorrow